src/hooks/usePayment.jsx

```diff
@@ -0,0 +1,148 @@
+  owner: "",
+  expiry: "",
+  nickname: "",
+  cvc: "",
+  password1: "",
+  password2: "",
+  backgroundColor: "#e5e5e5",
+};
+
+const digitTypeInput = ["cvc", "password1", "password2"];
+
+const formatDigitTypeInput = ["number", "expiry"];
+
+const cardCompanyList = [
+  {
+    company: "포코카드",
+    backgroundColor: "red",
+  },
+  {
+    company: "준카드",
+    backgroundColor: "palegreen",
+  },
+  {
+    company: "혜원카드",
+    backgroundColor: "skyblue",
+  },
+  {
+    company: "윤호카드",
+    backgroundColor: "yellow",
+  },
+  // {
+  //   company: "다빈카드",
+  //   backgroundColor: "pink",
+  // },
+  // {
+  //   company: "나나카드",
+  //   backgroundColor: "yellowgreen",
+  // },
+  // {
+  //   company: "치치카드",
+  //   backgroundColor: "plum",
+  // },
+  // {
+  //   company: "사나카드",
+  //   backgroundColor: "lightseagreen",
+  // },
+  // {
+  //   company: "랄랄카드",
+  //   backgroundColor: "burlywood",
+  // },
+];
+
+const usePayment = () => {
+  const [cardInfo, setCardInfo] = useState(defaultCardInfo);
+  const [cardList, setCardList] = useState([]);
+  const [isShowPopup, setIsShowPopup] = useState(true);
+
+  const navigate = useNavigate();
+
+  const onHandleCardInfoInput = evt => {
+    let { value, id } = evt.target;
+    if (!checkValidInputValue(value, id))
+      value = value.subString(0, value.length - 1);
+    setCardInfo({ ...cardInfo, [id]: formatInputValue(value, id) });
+  };
+
+  const onHandleResetCardInfo = () => {
+    setCardInfo(defaultCardInfo);
+  };
+
+  const onHandleSubmit = evt => {
+    evt.preventDefault();
+    navigate("/save");
+  };
+
+  const checkValidInputValue = (value, id) => {
+    if (digitTypeInput.indexOf(id) !== -1 && value.match(/[^0-9]/)) {
+      alert("숫자만 입력가능합니다.");
+      return false;
+    }
+    return true;
+  };
+
+  const onHandleCompanyPopupClick = evt => {
+    const clickedElement = evt.target.closest(".modal-item-container");
+    if (clickedElement) {
+      const { id } = clickedElement;
+      const [owner, backgroundColor] = id.split("-");
+      setCardInfo({
+        ...cardInfo,
+        owner,
+        backgroundColor,
+      });
+    }
+    setIsShowPopup(false);
+  };
+
+  const onHandleSave = () => {
+    setCardList([...cardList, cardInfo]);
+    navigate("/");
+  };
+
+  const formatInputValue = (value, id) => {
+    let formattedValue = value;
+
+    if (formatDigitTypeInput.indexOf(id) === -1) return value;
+    if (id === "number") {
+      const regex = /[^0-9]/g;
+      const cleanInput = value.replace(regex, "");
+      const chunks = cleanInput.match(/.{1,4}/g);
+      if (!chunks) {
+        return cleanInput;
+      }
+      formattedValue = chunks.join("-");
+    }
+    if (id === "expiry") {
+      const regex = /[^0-9]/g;
+      const cleanInput = value.replace(regex, "");
+      const chunks = cleanInput.match(/.{1,2}/g);
+      if (!chunks) {
+        return cleanInput;
+      }
+      formattedValue = chunks.join("/");
+    }
+    return formattedValue;
```

Comment:
함수를 이해하기 쉽게 추상화해보면 어떨까요?
예를들어..

```suggestion
    // 현재는 value가 필요없지만 예시로 넣어뒀어요!
    const isFormatRequired = checkRequireFormatting({ type, value });
    
    if (!isFormatRequired) {
       return value;
    }

    return getFormatedValue({ type, value });
}

function getFormatedValue({ type, value }) {
    switch(type) {
        case "expiry":
            return formatExpiryValue({ type, value });
        // ...
    }
    // or look up 테이블에 key로 type, value로 함수를 가지게 만들어서 바로 return 해도 좋을 것 같아요
}
```